From c4c7c0ccd26836a2a60ef5d424935eb469151acd Mon Sep 17 00:00:00 2001
From: Andy McCrae <andy.mccrae@gmail.com>
Date: Tue, 8 Dec 2015 12:02:22 +0000
Subject: [PATCH] Fix metric for rabbitmq sockets MaaS alarms

This fixes an error where the metric were listed as:
metric["rabbitmq_socket_used"] & metric["rabbitmq_socket_total"]
These should be fixed to be "sockets" rather than "socket".

Closes Bug: #1522897
Change-Id: I63ae2be2a09d2b9fa04c60e48805e42b8f3a5874
---
 rpc_deployment/playbooks/monitoring/maas_local.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rpc_deployment/playbooks/monitoring/maas_local.yml b/rpc_deployment/playbooks/monitoring/maas_local.yml
index 94af0c9..6b3f2db 100644
--- a/rpc_deployment/playbooks/monitoring/maas_local.yml
+++ b/rpc_deployment/playbooks/monitoring/maas_local.yml
@@ -325,7 +325,7 @@
       - { 'name': 'rabbitmq_max_channels_per_conn', 'criteria': ':set consecutiveCount={{ maas_alarm_local_consecutive_count }} if (metric["rabbitmq_max_channels_per_conn"] > 10) { return new AlarmStatus(CRITICAL, "Detected RabbitMQ connections with > 10 channels, check RabbitMQ and all Openstack consumers"); }' }
       - { 'name': 'rabbitmq_fd_used_alarm_status', 'criteria': ':set consecutiveCount={{ maas_alarm_local_consecutive_count }} if (percentage(metric["rabbitmq_fd_used"],metric["rabbitmq_fd_total"]) >= 90 ) { return new AlarmStatus(CRITICAL, "RabbitMQ file descriptors is reaching configured limit"); }' }
       - { 'name': 'rabbitmq_proc_used_alarm_status', 'criteria': ':set consecutiveCount={{ maas_alarm_local_consecutive_count }} if (percentage(metric["rabbitmq_proc_used"],metric["rabbitmq_proc_total"]) >= 90 ) { return new AlarmStatus(CRITICAL, "RabbitMQ processes is reaching configured limit"); }' }
-      - { 'name': 'rabbitmq_socket_used_alarm_status', 'criteria': ':set consecutiveCount={{ maas_alarm_local_consecutive_count }} if (percentage(metric["rabbitmq_socket_used"],metric["rabbitmq_socket_total"]) >= 90 ) { return new AlarmStatus(CRITICAL, "RabbitMQ sockets is reaching configured limit"); }' }
+      - { 'name': 'rabbitmq_socket_used_alarm_status', 'criteria': ':set consecutiveCount={{ maas_alarm_local_consecutive_count }} if (percentage(metric["rabbitmq_sockets_used"],metric["rabbitmq_sockets_total"]) >= 90 ) { return new AlarmStatus(CRITICAL, "RabbitMQ used sockets is reaching configured limit"); }' }
       - { 'name': 'rabbitmq_msgs_excl_notifications', 'criteria': ' :set consecutiveCount={{ maas_alarm_local_consecutive_count }} if (metric["rabbitmq_msgs_excl_notifications"] > {{ rabbitmq_queued_messages_excluding_notifications_threshold }} ) { return new AlarmStatus(CRITICAL, "RabbitMQ sum of queued messages excluding notifications queues is reaching configured limit. Currently above {{ rabbitmq_queued_messages_excluding_notifications_threshold }}"); }'}
       - { 'name': 'rabbitmq_qgrowth_excl_notifications', 'criteria': ':set consecutiveCount={{ maas_alarm_local_consecutive_count }} if (rate(metric["rabbitmq_msgs_excl_notifications"]) > {{rabbitmq_queue_growth_rate_threshold / maas_check_period}}) { return new AlarmStatus(CRITICAL, "RabbitMQ Queue growth rate is above configured threshold. Currently above {{ rabbitmq_queue_growth_rate_threshold }}"); }'}
   user: root
-- 
2.6.4

